### PR TITLE
List no-dependenciy tasks and assign to user

### PR DESCRIPTION
**Pull request by vitorschelb**
#50 

This PR creates a list no-dependenciy tasks and assign to user

- Ref dependency tasks on seed
-  Create class TasksDAO on tasks.dao.ts with method to look for all tasks that don't depend on other tasks.
- Creates method on DB to assing these tasks to the user (UserTasks table)
-  Assign the no-dependency tasks to the user on the pos onboarding route flow

### CODE REVIEW

File: src/app/api/onboarding/route.ts [hunk 1]
```diff
@@ -15,8 +16,14 @@ export async function POST(request: NextRequest) {
 
     const session = await getServerSession(authOptions);
     const email = session?.user?.email ?? raise("E-mail not available");
+    const userData =
+      (await user.fetchUserBy({ email })) ?? raise("User not found");
+
     await user.updateUserOnboardingBy({ email }, data);
 
+    const tasksWithNoDependencies = await task.listNoDependenciesTasks();
+    await user.assignTasksToUser(userData?.id, tasksWithNoDependencies);
```
Comment (marco-souza) [inline src/app/api/onboarding/route.ts:25]:
```suggestion
    await user.assignTasksToUser(userData.id, tasksWithNoDependencies);
```

File: src/shared/db.ts [hunk 2]
```diff
@@ -61,6 +61,18 @@ class UserDAO {
     });
   }
 
+  async assignTasksToUser(userId: number, tasks: Task[]) {
+    const tasksData = tasks.map((task) => ({
+      userId: userId,
```
Comment (marco-souza) [inline src/shared/db.ts:66]:
```suggestion
      userId,
```

**Review by marco-souza — APPROVED** [review]
Nice one! 

I left some minor style comments, but LGTM!

**Comment by marco-souza** [comment]
we still need to resolve the integration test before merging it

**Comment by marco-souza** [comment]
Also, friendly reminder to update the PR title, to something more readable

File: src/tests/db.test.ts [hunk 3]
```diff
@@ -42,7 +42,7 @@ describe("user test cases", () => {
         where: { taskId: t.id },
       });
 
-      expect(userTasks.length).toBeGreaterThan(0);
+      expect(userTasks.length);
```
Comment (marco-souza) [inline src/tests/db.test.ts:45]: I think you removed the assert here

```suggestion
      expect(userTasks.length).toHaveLength(0);
```

**Review by marco-souza — CHANGES_REQUESTED** [review]

File: src/tests/db.test.ts (hunk 3 above)
Comment (vitorschelb) [reply src/tests/db.test.ts]: expect(userTasks).toHaveLength(0)

**Review by marco-souza — APPROVED** [review]
The Way

**Pull request merged**